File: api/src/main/java/org/delivery/api/domain/store/controller/StoreApiController.java
package org.delivery.api.domain.store.controller;

import lombok.RequiredArgsConstructor;
import org.delivery.common.api.Api;
import org.delivery.api.domain.store.business.StoreBusiness;
import org.delivery.api.domain.store.controller.model.StoreResponse;
import org.delivery.db.store.enums.StoreCategory;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;

@RestController
@RequiredArgsConstructor
@RequestMapping("/api/store")
public class StoreApiController {

    private final StoreBusiness storeBusiness;
    @GetMapping("/search")
    public Api<List<StoreResponse>> search(
            @RequestParam(required = false)
            StoreCategory category
    ){
        var response = storeBusiness.searchCategory(category);
        return Api.OK(response);
    }
}

File: api/src/main/kotlin/org/delivery/api/domain/store/controller/StoreApiController.kt
package org.delivery.api.domain.store.controller

import org.delivery.api.domain.store.business.StoreBusiness
import org.delivery.api.domain.store.controller.model.StoreResponse
import org.delivery.common.api.Api
import org.delivery.db.store.enums.StoreCategory
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RequestParam
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/api/store")
class StoreApiController(
    val storeBusiness: StoreBusiness
){
    @GetMapping("/search")
    fun search(
        @RequestParam(required = false)
        category:StoreCategory
    ):Api<List<StoreResponse>>{
        storeBusiness.searchCategory(category)
            .let{
                return Api.OK(body=it)
            }
    }

}